models/demos/t3000/llama2_70b/tt/llama_attention_optimized.py

```diff
@@ -465,30 +476,38 @@ def prefill_attn_selfout(self, attn_output):
         # ATTENTION SELFOUT
         attn_output = ttnn.experimental.nlp_concat_heads(
             attn_output,
-            memory_config=self.model_config["DRAM_MEMCFG"],
+            memory_config=ttnn.DRAM_MEMORY_CONFIG,
         )  # seqlen, 1, batch, hidden_size
 
         attn_output = ttnn.all_gather(
             attn_output,
             dim=3,
             num_links=self.model_config["ALL_GATHER_NUM_LINKS"],
-            memory_config=self.model_config["DRAM_MEMCFG"],
+            memory_config=ttnn.DRAM_MEMORY_CONFIG,
         )
 
         _, _, seq_len, _ = attn_output.shape
-
         max_mm_seq_len = self.model_config["MAX_MM_SEQ_LEN"]
-        batch_dim = 1 if seq_len < max_mm_seq_len else seq_len // max_mm_seq_len  # Find the division factor
-        attn_output = ttnn.reshape(attn_output, (1, batch_dim, seq_len // batch_dim, -1))
+        if seq_len >= max_mm_seq_len:
+            if seq_len % max_mm_seq_len != 0:
+                raise ValueError(f"seq_len {seq_len} must be divisible by {max_mm_seq_len}")
+            batch_dim = 1 if seq_len < max_mm_seq_len else seq_len // max_mm_seq_len  # Find the division factor
```

Comment:
The else will always be hit here due the outer if. Same comment for `if seq_len >= max_mm_seq_len:` cases in other files.

Reply:
in Llama, `max_mm_seq_len=1024`, so seq_len > 1024 and multiple of 1024 should enter this top if